pints/_mcmc/__init__.py

```diff
@@ -774,16 +830,30 @@ def run(self):
             if self._log_to_screen:
                 print(halt_message)
 
-        # Store generated chains in memory
-        if self._chains_in_memory:
-            self._samples = samples
-
         # Store evaluations in memory
         if self._evaluations_in_memory:
             self._evaluations = evaluations
 
+        # Inverse transform to model space if transform is provided
+        if self._chains_in_memory:
+            if self._transform:
+                n_c, n_s, n_p = samples.shape
+                samples_user = np.zeros((n_c, n_s, n_p))
+                for c in range(n_c):
+                    for s in range(n_s):
+                        samples_user[c, s, :] = \
+                            self._transform.to_model(samples[c, s, :])
+            else:
```

Comment:
I'd get rid of this and write `samples = samples_user` in the above, followed by del(samples_user)

Reply:
Done